### PR TITLE
Bugfix/pp9 14855 get rid of lint warnings

### PR DESCRIPTION
**Pull request by Grim--Reaper**

### CODE REVIEW

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-oidc/src/lib/picturepark-oidc.module.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ export function storageFactory(): OAuthStorage {
   ],
 })
 export class PictureparkOidcModule {
+  // eslint-disable-next-line @typescript-eslint/ban-types
   static forRoot(config: PictureparkOidcAuthConfiguration | Function): ModuleWithProviders<PictureparkOidcModule> {
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-oidc/src/lib/picturepark-oidc.module.ts:23]: try to get rid of the disable instruction (change type Function)

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/content-browser/components/content-image-preview/content-image-preview.component.ts [hunk 2]
```diff
@@ -266,20 +270,16 @@ export class ContentImagePreviewComponent extends BaseComponent implements OnIni
...
+              s.outputs.find(i => i.outputFormatId === 'VideoLarge')?.downloadUrl ??
+              s.outputs.find(i => i.outputFormatId === 'VideoSmall')?.downloadUrl,
+            audioUrl: s.outputs.find(i => i.outputFormatId === 'AudioSmall')?.viewUrl,
             outputs: s.outputs,
           } as IShareItem;
         }),
       };
 
-      item = share.items.find(i => i.id === this.content.id)!;
+      item = share.items.find(i => i.id === this.content.id) ?? ({} as IShareItem);
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/content-browser/components/content-image-preview/content-image-preview.component.ts:282]: only as IShareItem

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/content-browser/content-browser.component.ts [hunk 3]
```diff
@@ -33,8 +33,7 @@ import { SessionService } from '../../shared-module/services/session/session.ser
   ],
 })
 export class ContentBrowserComponent extends BaseBrowserComponent<Content> implements OnChanges {
-  @Input()
-  channel: Channel | null = null;
+  @Input() channel: Channel;
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/content-browser/content-browser.component.ts:36]: revert this change

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/content-browser/content-browser.component.ts [hunk 4]
```diff
@@ -106,16 +105,13 @@ export class ContentBrowserComponent extends BaseBrowserComponent<Content> imple
...
 
-      this.facade.searchRequestState.channelId = this.channel!.id;
+      this.facade.searchRequestState.channelId = channel.id;
       // Trigger load
-      if (this.channel?.aggregations) {
-        this.facade.patchRequestState({ aggregators: this.channel.aggregations });
-      } else {
-        this.facade.patchRequestState({});
-      }
+      this.facade.patchRequestState({ aggregators: channel.aggregations });
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/content-browser/content-browser.component.ts:114]: Check whether the logic stayed identical, otherwise revert

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts [hunk 5]
```diff
@@ -99,6 +99,7 @@ export class ContentDownloadDialogComponent extends DialogBaseComponent implemen
         .subscribe(output => this.setSelection(output));
     } else {
       if (this.data.contents.every(content => content.outputs)) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
         const outputs = flatMap(this.data.contents, content => content.outputs!);
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:103]: try to pass an empty array in case the outputs were undefined and delete the disable

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/layer-panels/services/layer-field.service.ts [hunk 6]
```diff
@@ -260,13 +260,13 @@ export class LayerFieldService {
       ContentResolveBehavior.OuterDisplayValueList,
     ]);
 
-    const relationFieldInfo = forkJoin([thumbnailDownload, contentDetail]).pipe(
-      map(response => {
+    const relationFieldInfo = forkJoin({ download: thumbnailDownload, content: contentDetail }).pipe(
+      map(joined => {
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/layer-panels/services/layer-field.service.ts:264]: try to create the objects in the map function

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/list-browser/list-browser.component.ts [hunk 7]
```diff
@@ -64,35 +64,37 @@ export class ListBrowserComponent extends BaseBrowserComponent<ListItem> impleme
...
-      const id = field.id.split('.').pop();
-      const names = field.names;
-      return { id, names, field };
-    });
-
-    this.displayedColumns = this.schema.fields!.map(field => {
-      const id = field.id.split('.').pop();
-      return id!;
-    });
+    this.displayedColumnNames =
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/list-browser/list-browser.component.ts:67]: revert the first part (variables)

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/list-items/list/list.component.ts [hunk 8]
```diff
@@ -66,10 +66,12 @@ export class ListComponent implements OnInit, OnDestroy {
...
     this.schema = this.route.paramMap.pipe(
-      flatMap(paramMap => {
-        const schemaId = paramMap.get('id')!;
+      mergeMap(paramMap => {
+        const schemaId = paramMap.get('id');
+        if (!schemaId) return of(undefined);
         return this.schemaService.get(schemaId);
-      })
+      }),
+      filter(s => !!s)
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/list-items/list/list.component.ts:74]: move before the mergMap

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-oidc/src/lib/picturepark-oidc.module.ts (hunk 1 above)
Comment (Grim--Reaper) [reply src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-oidc/src/lib/picturepark-oidc.module.ts:23]: the useFactory property type is also defined as Function

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/shared-module/services/metadata-preview/metadata-preview.service.ts [hunk 9]
```diff
@@ -130,7 +135,7 @@ export class MetaDataPreviewService {
             return v.join(', ');
           })
           .join(', ');
-        console.log(fields[fieldId]);
+        this.logger.debug(fields[fieldId]);
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/shared-module/services/metadata-preview/metadata-preview.service.ts:138]: remove logging

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/shared-module/services/metadata-preview/metadata-preview.service.ts [hunk 10]
```diff
@@ -163,7 +168,7 @@ export class MetaDataPreviewService {
       } else if (fieldType === FieldMultiRelation) {
         // Unsupported
       } else {
-        console.log('Unsupported field type [' + fieldType + '] encountered.');
+        this.logger.debug('Unsupported field type [' + fieldType + '] encountered.');
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/shared-module/services/metadata-preview/metadata-preview.service.ts:171]: log as error

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/content-search.facade.ts [hunk 11]
```diff
@@ -36,14 +36,14 @@ export class ContentSearchFacade extends SearchFacade<Content, ContentSearchInpu
     return this.contentService.search(request).pipe(finalize(() => this.setLoading(false)));
   }
 
-  searchAggregations(aggregators: AggregatorBase[]): Observable<AggregationResult[]> | undefined {
+  searchAggregations(aggregators: AggregatorBase[]): Observable<AggregationResult[]> {
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/content-search.facade.ts:39]: remove the return types in the whole file

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/list-item-search.facade.ts [hunk 12]
```diff
@@ -33,10 +33,10 @@ export class ListItemSearchFacade extends SearchFacade<ListItem, ListItemSearchI
     return this.listItemService.search(request).pipe(tap(() => this.setLoading(false)));
   }
 
-  searchAggregations(aggregators: AggregatorBase[]): Observable<AggregationResult[]> | undefined {
+  searchAggregations(aggregators: AggregatorBase[]): Observable<AggregationResult[]> {
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/list-item-search.facade.ts:36]: check the return types and remove if not needed

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/schema-search.facade.ts [hunk 13]
```diff
@@ -61,7 +61,7 @@ export class SchemaSearchFacade extends SearchFacade<Schema, SchemaSearchInputSt
     return this.schemaService.search(request).pipe(tap(() => this.setLoading(false)));
   }
 
-  searchAggregations(aggregators: AggregatorBase[]): Observable<AggregationResult[]> | undefined {
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/schema-search.facade.ts:64]: check the return types and remove if not needed

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/schema-search.facade.ts [hunk 14]
```diff
@@ -61,7 +61,7 @@ export class SchemaSearchFacade extends SearchFacade<Schema, SchemaSearchInputSt
...
-    throw new Error('Method not implemented.');
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/schema-search.facade.ts:65]: Revert this change and throw an exception

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/search.facade.ts [hunk 15]
```diff
@@ -94,7 +94,8 @@ export abstract class SearchFacade<T, TState extends SearchInputState> {
       }>
     | undefined;
 
-  abstract searchAggregations(aggregators: AggregatorBase[]): Observable<AggregationResult[]> | undefined;
+  // TODO WIM (PP9-14855): ensure that this change (remove undefined return type) still works in any case
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/search.facade.ts:97]: check the TODO and remove comment if everything ok. Check where it's used

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/share-search.facade.ts [hunk 16]
```diff
@@ -26,10 +26,10 @@ export class ShareSearchFacade extends SearchFacade<Share, SearchInputState> {
     return this.shareService.search(request).pipe(tap(() => this.setLoading(false)));
   }
 
-  searchAggregations(aggregators: AggregatorBase[]): Observable<AggregationResult[]> | undefined {
+  searchAggregations(aggregators: AggregatorBase[]): Observable<AggregationResult[]> {
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/facade/share-search.facade.ts:29]: check the return types and remove if not needed

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/locale.module.ts [hunk 17]
```diff
@@ -20,7 +21,7 @@ export function loadLanguagesFactory(
 ): () => Promise<boolean | undefined> {
   const languageToSelect = language || getLocaleFactory(localStorageService);
   return () => {
-    return languageService.loadLanguages(allowedLanguages, languageToSelect, cdnUrl).toPromise();
+    return lastValueFrom(languageService.loadLanguages(allowedLanguages, languageToSelect, cdnUrl));
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/locale.module.ts:24]: add take(1) to ensure it's properly disposed

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/services/noop-auth.service.ts [hunk 18]
```diff
@@ -9,10 +9,10 @@ import { AuthService } from './auth.service';
 export class NoopAuthService extends AuthService {
   constructor(
     @Inject(LOCALE_ID) private locale: string,
-    @Optional() @Inject(PICTUREPARK_API_URL) pictureparkApiUrl?: string,
+    @Optional() @Inject(PICTUREPARK_API_URL) pictureparkApiUrl: string,
     @Optional() @Inject(PICTUREPARK_CONFIGURATION) pictureparkConfiguration?: PictureparkConfiguration
   ) {
-    super(pictureparkConfiguration?.apiServer || pictureparkApiUrl!);
+    super(pictureparkConfiguration?.apiServer ?? pictureparkApiUrl);
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/services/noop-auth.service.ts:15]: revert

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/tests/channel-service.spec.ts [hunk 19]
```diff
@@ -4,38 +4,33 @@ import { inject, waitForAsync } from '@angular/core/testing';
...
-        expect(channel?.names.translate('en-US')).toBe('Root Channel');
-      })
-    )
-  );
+  it('should response with at least one channel', waitForAsync(
+    inject([AuthService, ChannelService], async (authService: AuthService, channelService: ChannelService) => {
+      // arrange
+
+      // act
+      const channels = await lastValueFrom(channelService.getAll());
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/tests/channel-service.spec.ts:17]: add take(1)

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/tests/content-service.spec.ts [hunk 20]
```diff
@@ -11,97 +11,88 @@ import {
   ContentAggregationRequest,
   TermsAggregator,
 } from '../lib/services/api-services';
+import { lastValueFrom } from 'rxjs';
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/tests/content-service.spec.ts:14]: add take(1)

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/tests/share-service.spec.ts [hunk 21]
```diff
@@ -11,55 +11,53 @@ import {
...
+      async (
+        contentService: ContentService,
+        shareService: ShareService,
+        businessProcessService: BusinessProcessService
+      ) => {
+        // arrange
+        const request = new ContentSearchRequest();
+        request.searchString = 'm';
+
+        const response = await lastValueFrom(contentService.search(request));
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/tests/share-service.spec.ts:31]: take(1)

File: src/picturepark-sdk-v2-angular/src/app/content-picker/embed.service.ts [hunk 22]
```diff
@@ -6,13 +6,19 @@ import {
   ShareContent,
   Content,
   BusinessProcessService,
+  LoggerService,
 } from '@picturepark/sdk-v2-angular';
+import { lastValueFrom } from 'rxjs';
```
Comment (Grim--Reaper) [inline src/picturepark-sdk-v2-angular/src/app/content-picker/embed.service.ts:11]: take(1)

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/list-items/list/list.component.ts [hunk 23]
```diff
@@ -126,7 +129,7 @@ export class ListComponent implements OnInit, OnDestroy {
   changeAggregationFilters(aggregationFilters: AggregationFilter[]): void {
     this.deselectSelectedItems();
     this.aggregationFilters = aggregationFilters;
-    const filtersQuery = this.aggregationFilters.map(filter => JSON.stringify(filter.toJSON()));
+    const filtersQuery = this.aggregationFilters.map(f => JSON.stringify(f.toJSON()));
```
[the remark this answers is not in the document]
Comment (Grim--Reaper) [reply src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular-ui/src/lib/features-module/list-items/list/list.component.ts:132]: 'filter' is already declared in the upper scope

File: src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/locale.module.ts (hunk 17 above)
Comment (Grim--Reaper) [reply src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/locale.module.ts:24]: No need to (and you can't pipe anything) as this doesn't return an observable but a promise.
According to this page (https://indepth.dev/posts/1287/rxjs-heads-up-topromise-is-being-deprecated) the now deprecated toPromise() function resolves once the observable completes. Therefore the lastValueFrom is the appropriate replacement as this one also resolves when the observable completes. firstValueFrom resolves on next() call

**Review by neotek — APPROVED** [review]
